core/src/Cuda/Kokkos_Cuda_KernelLaunch.hpp

```diff
@@ -86,6 +86,15 @@ inline __device__ T* kokkos_impl_cuda_shared_memory() {
 
 namespace Kokkos {
 namespace Impl {
+// CachePreferences during kernel launch.
+// KokkosCachePreferL1 = 0 - cudaFuncCachePreferL1
+// KokkosCachePreferShared = 1 - cudaFuncCachePreferShared
+// KokkosCachePreferEqual = 2 - cudaFuncCachePreferEqual
+enum CachePreference {
+  KokkosCachePreferL1     = 0,
+  KokkosCachePreferShared = 1,
+  KokkosCachePreferEqual  = 2
```

Comment:
Is there a particular reason to number them explicitly?

Reply:
No particular reason. Just gave it values when implementing the logic. Will that be an issue?

Reply:
This is a hack so that the `prefer_shmem` boolean argument in the kernel launch is interpreted in a backward compatible way.  We requested to use a scoped enumeration which means that the call sites will need to be updated and that the values will be irrelevant.